merino/providers/top_picks/provider.py

```diff
@@ -52,14 +52,18 @@ async def initialize(self) -> None:
     def hidden(self) -> bool:  # noqa: D102
         return False
 
+    def normalize_query(self, query: str) -> str:
+        """Convert a query string to lowercase and remove trailing spaces."""
+        return query.strip().lower()
+
     async def query(self, srequest: SuggestionRequest) -> list[BaseSuggestion]:
         """Query Top Pick data and return suggestions."""
         # Ignore https:// and http://
         if srequest.query.startswith("http"):
             return []
 
         qlen: int = len(srequest.query)
-        query: str = srequest.query
+        query: str = self.normalize_query(srequest.query)
```

Comment:
nitpick: Ditto here.

```suggestion
        query: str = srequest.query
```